File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length <= 1)
            return nums.length;

        HashSet<Integer> allNums = new HashSet<>();
        for (int num : nums) {
            allNums.add(num);
        }

        int maxSeqLen = 0;

        for (int num : nums) {
            if (!allNums.contains(num - 1)) 
            {
                int thisSeqLen = 1;
                int thisNum = num + 1;
                while (allNums.contains(thisNum)) {
                    thisSeqLen++;
                    thisNum++;
                }
                maxSeqLen = Math.max(maxSeqLen, thisSeqLen);
            }
        }
        return maxSeqLen;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution(object):

    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maxSeq = 0
        numSet = set(nums)

        for num in nums:
            if (num-1 not in numSet):
                thisSeq = 1
                while (num+thisSeq in numSet):
                    thisSeq+=1
                maxSeq=max(thisSeq, maxSeq)

        return maxSeq

        